probes/blocksDeleteOnBranches/def.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: blocksDeleteOnBranches
+short: Check that the project blocks non-admins from deleting branches.
+motivation: >
+  Allowing non-admins to delete project branches could prevent users from consuming the project.
+implementation: >
+  Checks the protection rules of each branch.
+outcome:
+  - The probe returns one OutcomePositive for each branch that is disallowed from users deleting it, and one OutcomeNegative for branches where users are able to delete it.
+remediation:
+  effort: High
+  text:
+    - Disallow deletion of branches in your project to remove negative outcomes.
```

Comment:
I think remediation effort is low-ish.  At least on GitHub it's pretty easy. Should the instructions mention any specifics on how to do so based on GitHub/GitLab?

Reply:
Improved in https://github.com/ossf/scorecard/pull/3691/commits/4071df489be2634f91b81db399f9f38a9cc70841